Core/include/Acts/Vertexing/AdaptiveGridDensityVertexFinder.hpp

```diff
@@ -0,0 +1,170 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "Acts/Vertexing/AdaptiveGridTrackDensity.hpp"
+#include "Acts/Vertexing/Vertex.hpp"
+#include "Acts/Vertexing/VertexingOptions.hpp"
+
+#include "DummyVertexFitter.hpp"
+
+namespace Acts {
+
+/// @class AdaptiveGridDensityVertexFinder
+/// @brief Vertex finder that makes use of a track density grid.
+/// Each single track is modelled as a 2(!)-dim Gaussian distribution grid
```

Comment:
```suggestion
/// Each single track is modelled as a 2-dim Gaussian distribution grid
```
you explicitly mention the relation to the 1D density vector, no need to add the (to me: confusing) exclamation mark.